### PR TITLE
auth_dbname option for database definitions

### PR DESCRIPTION
**Pull request by dbonne**
Now you can specify an database's entry name for authentication.
If auth_dbname is set, auth_query is sended to that database.
This is util were we want use only one database for all users
authentications, i.e

   p0 = host=127.0.0.1 dbname=pgbench auth_dbname=authdb auth_user=pswcheck
   authdb = host=127.0.0.1 dbname=pgbouncer

If there isn't an entry named as the value specified in auth_dbname,
the database specified in dbname will be used instead.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,9 @@
 /test/test.log
 /test/test.pid
 
+# Visual Studio Code
```
Comment (petere) [inline .gitignore]: remove

File: test/test.sh [hunk 2]
```diff
@@ -49,9 +49,9 @@ if test -n "$CAN_SUDO"; then
 		sudo pfctl -a pgbouncer -F all -q 2>&1 | grep -q "pfctl:" && {
 			cat <<-EOF
 			Please enable PF and add the following rule to /etc/pf.conf
-
```
Comment (petere) [inline test/test.sh]: remove or make separate commit

File: test/test.sh [hunk 3]
```diff
@@ -75,8 +75,8 @@ stopit test.pid
 stopit pgdata/postmaster.pid
 
 mkdir -p $LOGDIR
-rm -f $BOUNCER_LOG $PG_LOG
-rm -rf $PGDATA
+echo > $BOUNCER_LOG
```
Comment (petere) [inline test/test.sh]: explain please

Comment (dbonne) [reply test/test.sh]: My mistake including this in the commit. I will fix this.

File: test/test.sh (hunk 2 above)
Comment (dbonne) [reply test/test.sh]: I should remove my branch and recreate it for create a separate commit? I'm not a git expert.

File: .gitignore (hunk 1 above)
Comment (dbonne) [reply .gitignore]: I will fix this.
